src/apps/AraConsole/AraConsole.js

```diff
@@ -0,0 +1,293 @@
+      )
+
+      const installAppIntent = [
+        [
+          kernelProxyAddress,
+          'newAppInstance(bytes32,address,bytes,bool)',
+          [appId, contractAddress, encodedInitializeFunc, false],
+        ],
+      ]
+
+      const aclProxyAddress = apps.find(app => app.name.toLowerCase() === 'acl')
+        .proxyAddress
+
+      const permissionIntents = permParams.map(([role, from, to]) => {
+        const roleBytes = roles.find(availableRole => availableRole.id === role)
+          .bytes
+
+        return [
+          aclProxyAddress,
+          'createPermission',
+          [to, counterfactualAppAddr, roleBytes, from],
+        ]
+      })
+
+      const intentBasket = [...installAppIntent, ...permissionIntents]
+      log(intentBasket, permissionIntents)
+      const {
+        path: pathForBasket,
+        transactions,
+      } = await wrapper.getTransactionPathForIntentBasket(intentBasket)
+      log('DONE', pathForBasket, transactions)
+      if (Array.isArray(pathForBasket) && pathForBasket.length) {
+        const transId = await wrapper.performTransactionPath(pathForBasket)
+        log(transId)
+      } else if (Array.isArray(transactions) && transactions.length) {
+        for (const transaction of transactions) {
+          await wrapper.performTransactionPath([transaction])
+        }
+      } else {
+        await wrapper.performTransactionPath([])
+      }
+    },
+    [toast, apps, wrapper]
+  )
+  // Handle DAO execution
+  const handleDaoExec = useCallback(
+    async params => {
+      if (params.length < 3) {
+        toast('Not enough arguments for DAO EXEC')
+        return
+      }
+      const [appName, methodSignature, ...args] = params
+      log('params', appName, methodSignature, args)
+      apps.find(app => {
+        console.log(`comparing ${app.name.toLowerCase()} to ${appName}`)
+        return app.name.toLowerCase() === appName
+      })
+
+      const proxyAddress = apps.find(app => app.name.toLowerCase() === appName)
+        .proxyAddress
+      const path = await wrapper.getTransactionPath(
+        proxyAddress,
+        methodSignature,
+        args
+      )
+
+      if (Array.isArray(path) && path.length) {
+        const transId = await wrapper.performTransactionPath(path)
+        log(transId)
+      } else {
+        await wrapper.performTransactionPath([])
+      }
+      log('trans path', path)
+    },
+    [apps, wrapper, toast]
+  )
+
+  const handleDaoAct = useCallback(() => {}, [])
+
+  // Handle console input
+  const handleConsoleInput = useCallback(() => {
+    if (command.includes('exec')) {
+      handleDaoExec(command.split(' ').slice(2))
+    } else if (command.includes('install')) {
+      handleDaoInstall(command.split(' ').slice(2))
+    } else if (command.includes('act')) {
+      handleDaoAct()
+    } else {
+      toast('Unrecognized Command')
+      setCommand('')
+    }
+  }, [command, handleDaoExec, handleDaoInstall, handleDaoAct, toast])
```

Comment:
`useCallback()` doesn’t provide much since we are not passing it to a pure component, so maybe we could make this part a bit easier to read :smile: 

```jsx
const handleConsoleInput = () => {
  if (command.includes('exec')) {
    handleDaoExec(command.split(' ').slice(2))
    return
  }

  if (command.includes('install')) {
    handleDaoInstall(command.split(' ').slice(2))
    return
  }

  if (command.includes('act')) {
    handleDaoAct()
    return
  }

  toast('Unrecognized Command')
  setCommand('')
}
```